libs/core/langchain_core/runnables/graph_mermaid.py

```diff
@@ -0,0 +1,211 @@
+
+    Args:
+        mermaid_syntax (str): Mermaid graph syntax. Generated using `draw_mermaid` method.
+        output_file (str, optional): Output file path. Defaults to "graph.png".
+        draw_method (MermaidDrawMethod, optional): Method to use for drawing the graph. The methods available and their usage are: 
+        - MermaidDrawMethod.API: Uses Mermaid.INK API to render the graph (default)
+        - MermaidDrawMethod.PYPPETEER: Uses Pyppeteer to render the graph
+
+    Example:
+        >>> node_ids = ['A', 'B', 'C', 'D']
+        >>> edges = [Edge('A', 'B'), Edge('B', 'C'), Edge('B', 'D'), Edge('A', 'D', 'end')]
+        >>> mermaid_syntax = draw_mermaid(node_ids, edges)
+        >>> draw_mermaid_png(mermaid_syntax, output_file='graph_api.png', draw_method=MermaidDrawMethod.API)
+        >>> # draw_mermaid_png(mermaid_syntax, output_file='graph_pyppeteer.png', draw_method=MermaidDrawMethod.PYPPETEER) # (Optionally if Pyppeteer is installed)
+    """
+    if draw_method == MermaidDrawMethod.PYPPETEER:
+        import asyncio
+        asyncio.run(_render_mermaid_using_pyppeteer(mermaid_syntax, output_file))
+    elif draw_method == MermaidDrawMethod.API:
+        _render_mermaid_using_api(mermaid_syntax,output_file)
+    else:
+        raise ValueError(f"Invalid draw method: {draw_method}. Supported draw methods are: {', '.join([m.value for m in MermaidDrawMethod])}")
+
+async def _render_mermaid_using_pyppeteer(mermaid_syntax: str, output_file: str):   
+    """Renders the Mermaid graph using Pyppeteer and saves it as a PNG image. It requires Pyppeteer to be installed. """ 
+    try:
+        from pyppeteer import launch
+    except ImportError:
+        raise ImportError("Install Pyppeteer to use the Pyppeteer method: `pip install pyppeteer`.")
+
+    browser = await launch()
+    page = await browser.newPage()
+
+    # Setup Mermaid JS
+    await page.goto('about:blank')
+    await page.addScriptTag({'url': 'https://unpkg.com/mermaid/dist/mermaid.min.js'})
+    await page.evaluate('''() => {
+                mermaid.initialize({startOnLoad:true});
+            }''')
+
+    # Render SVG
+    svg_code = await page.evaluate(f"""(mermaidGraph) => {{
+                return mermaid.mermaidAPI.render('mermaid', mermaidGraph);
+            }}""", mermaid_syntax)
+
+    # Set the page background to white
+    await page.evaluate('''(svg) => {
+            document.body.innerHTML = svg;
+            document.body.style.background = 'white'; // Set the page background to white
+        }''', svg_code['svg'])
+
+    # Take a screenshot
+    dimensions = await page.evaluate('''() => {
+            const svgElement = document.querySelector('svg');
+            const rect = svgElement.getBoundingClientRect();
+            return { width: rect.width, height: rect.height };
+        }''')
+    padding = 10  # Reduce padding to 10px or to a value that works for your case
```

Comment:
worth making configurable>?

Reply:
Good point, added both that and the background color